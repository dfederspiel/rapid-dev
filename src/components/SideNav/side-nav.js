import React from 'react';
import FontAwesome from 'react-fontawesome';

export default class SideNav extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            open: true,
            minimal: false,
            mobile: props.mobile
        };
    }

    componentWillReceiveProps(props) {
        this.setState({ open: props.open, mobile: props.mobile, minimal: props.minimal });
    }

    getClasses() {
        var classes = ['side-nav'];
        if (this.state.open)
            classes.push('open');
        if (this.state.mobile)
            classes.push('mobile');
        if (this.state.minimal)
            classes.push('minimal');

        return classes.join(' ');
    }

    getUserLinks = () => {
        var links = [
            { icon: 'star', href: '/', title: 'Some Link', active: true },
            { icon: 'star', href: '/', title: 'Some Link' },
            { icon: 'star', href: '/', title: 'Some Link' },
            { icon: 'star', href: '/', title: 'Some Link' },
            { icon: 'star', href: '/', title: 'Some Link' },
            { icon: 'star', href: '/', title: 'Some Link' },
            { icon: 'star', href: '/', title: 'Some Link' }
        ];

        let linkHtml = [];
        for (var x = 0; x < links.length; x++) {
            linkHtml.push(<a className={links[x].active ? 'active' : '' } key={x}><FontAwesome name='star' size="lg" />{!this.state.minimal && <label>TESTING</label>}</a>);
        }

        return linkHtml;
    }

    render() {
        return (
            <div className={this.getClasses()}>
                <div className="side-nav-upper">
                    <a className="active"><FontAwesome style={{ color: 'gray' }} name='star' size="lg" />{!this.state.minimal && <label>Home Overview</label>}</a>
                    <a><FontAwesome style={{ color: 'gray' }} name='bell' size="lg" />{!this.state.minimal && <label>Transactions</label>}</a>
                    <a><FontAwesome style={{ color: 'gray' }} name='bath' size="lg" />{!this.state.minimal && <label>Liquidity</label>}</a>
                    <a><FontAwesome style={{ color: 'gray' }} name='bed' size="lg" />{!this.state.minimal && <label>Investments</label>}</a>
                </div>
                <div className="side-nav-lower">
                    <label>FastTrack</label>
                    {this.getUserLinks()}
                </div>
            </div>
        );
    }
}