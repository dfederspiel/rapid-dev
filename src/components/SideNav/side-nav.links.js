
export function transactionsLinks() {
    return [
        { title: "Transaction Review", href:"/" },
        { title: "Cash Management Account", href:"/" },
        { title: "International Wires", href:"/" },
        {
            title: "Share Draft Services", href:"",
            sublinks:
                [
                    { title: "Third Level 1", href:"/" },
                    { title: "Third Level 2", href:"/" },
                    { title: "Third Level 3", href:"/" },
                ]
        },
        { title: "Check Deposit Services", href:"/" },
    ]
}
export function liquidityLinks() {
    return [
        { title: "Liquidity Link 1", href:"/" },
        { title: "Liquidity Link 2", href:"/" },
        { title: "Liquidity Link 3", href:"/" },
        { title: "Liquidity Link 4", href:"/" },
        { title: "Liquidity Link 5", href:"/" },
        { title: "Liquidity Link 6", href:"/" },
    ]
}

export function investmentsLinks() {
    return [
        { title: "Investments Link 1", href:"/" },
        { title: "Investments Link 2", href:"/" },
        { title: "Investments Link 3", href:"/" },
        { title: "Investments Link 4", href:"/" },
        { title: "Investments Link 5", href:"/" },
        { title: "Investments Link 6", href:"/" },
    ]
}