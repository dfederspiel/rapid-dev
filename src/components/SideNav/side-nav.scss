@import '../../styles/variables';

.side-nav {
    z-index: 100;
    display: none;
    position: absolute;
    @include at(720) {
        position: static
    }
    flex-direction: column;
    align-items: center;
    border-right: .1rem solid $border-light;
    border-bottom: .1rem solid  $border-light;
    background-color: $white;
    min-height: 100vh;

    .side-nav-upper {
        border-bottom: .1rem solid  $border-light;
        a {
            margin: .5rem auto;
        
            &:first-child {
                margin-top: 1rem;
            }
            &:last-child{
                margin-bottom: 3rem;
            }
        }
    }

    .side-nav-lower {
        
        > label {
            display: block;
            width: 100%;
            margin-top: 2rem;
            margin-left: 3.6rem;
        }
        
        a {
            margin: .25rem auto;
            &:first-child {
                margin-top: .5rem;
            }
            &:last-child{
                margin-bottom: 3rem;
            }
        }
    }

    .side-nav-lower, .side-nav-upper {
        display: flex;
        flex-direction: column;
        align-items: center;
        width: 100%;

        a {
            padding: .2rem .9rem;
            display: flex;
            align-items: center;
            width: 100%;
            min-height: 3.6rem;

            &.active, &:hover {
                padding: .2rem 0 .2rem .6rem;
                border-left: solid .3rem $primary;
                .fa {
                    color: $primary !important;
                }
                label {
                    font-weight: 600;
                }
            }

            .fa {
                display: flex;
                justify-content: center;
                align-items: center;
                width: 3.2rem;
                font-size: 1.5rem;
                margin-right: .8rem;
            }

            label {
                margin: 0;
                font-weight: 400;
            }
        }
    }

    &.open {
        display: flex;
        width: $side-nav-width;

        &.minimal {
            width: 5rem;

            label {
                display: none;
            }

            a {
                .fa {
                    margin-right: 0;
                }
            }
        }
    }
}
