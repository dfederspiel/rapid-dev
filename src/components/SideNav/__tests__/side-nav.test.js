import React from 'react';
import SideNav from '../side-nav';
import renderer from 'react-test-renderer';
import shallow from 'enzyme';

describe('the side navigation component', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <SideNav />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe("in mobile", () => {
        const nav = shallow(<SideNav open={false} />);
        expect(nav.contains(<div></div>)).toBe(true);
    });
});