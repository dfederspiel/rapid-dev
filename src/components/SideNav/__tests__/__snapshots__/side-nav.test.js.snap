// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the side navigation component renders correctly 1`] = `
<div
  className="side-nav"
>
  <div
    className="side-nav-upper"
  >
    <a>
      <span
        aria-hidden={true}
        className="fa fa-star fa-lg"
        style={
          Object {
            "color": "gray",
          }
        }
      />
    </a>
    <a>
      <span
        aria-hidden={true}
        className="fa fa-bell fa-lg"
        style={
          Object {
            "color": "gray",
          }
        }
      />
    </a>
    <a>
      <span
        aria-hidden={true}
        className="fa fa-bath fa-lg"
        style={
          Object {
            "color": "gray",
          }
        }
      />
    </a>
    <a>
      <span
        aria-hidden={true}
        className="fa fa-bed fa-lg"
        style={
          Object {
            "color": "gray",
          }
        }
      />
    </a>
  </div>
  <div
    className="side-nav-lower"
  >
    <a>
      <span
        aria-hidden={true}
        className="fa fa-star fa-lg"
        style={
          Object {
            "color": "gray",
          }
        }
      />
    </a>
    <a>
      <span
        aria-hidden={true}
        className="fa fa-bell fa-lg"
        style={
          Object {
            "color": "gray",
          }
        }
      />
    </a>
    <a>
      <span
        aria-hidden={true}
        className="fa fa-bath fa-lg"
        style={
          Object {
            "color": "gray",
          }
        }
      />
    </a>
    <a>
      <span
        aria-hidden={true}
        className="fa fa-bed fa-lg"
        style={
          Object {
            "color": "gray",
          }
        }
      />
    </a>
  </div>
</div>
`;
